--- src/configure.ac.1	2018-02-03 15:51:41.444206209 +0200
+++ src/configure.ac	2018-02-03 15:52:41.532737870 +0200
@@ -71,6 +71,25 @@
 ]], [])],
     [ac_cv_boost_python=yes], [ac_cv_boost_python=no])])
 
+#Overcome missing rpc.h with > glibc-2.26
+AC_ARG_WITH([libtirpc],
+      [AS_HELP_STRING([--with-libtirpc], [Use libtirpc as RPC implementation (instead of sunrpc)])])
+
+AS_IF([test "x$with_libtirpc" = xyes], 
+      [PKG_CHECK_MODULES([TIRPC], 
+                         [libtirpc], 
+                         [RPC_CFLAGS=$TIRPC_CFLAGS; RPC_LIBS=$TIRPC_LIBS;],
+                        [AC_MSG_ERROR([libtirpc requested, but library not found.])]
+                       )], 
+      [AC_CHECK_HEADER(rpc/rpc.h, 
+                      [RPC_CFLAGS=""; RPC_LIBS="";],
+                      [AC_MSG_ERROR([sunrpc requested, but headers are not present.])]
+                     )]
+)
+
+AC_SUBST(RPC_CFLAGS)
+AC_SUBST(RPC_LIBS)
+
 AS_IF([test "x$ac_cv_boost_python" = xyes],
 [
   BOOST_PYTHON_LIBS=
