From dad8975a8757401c681258a9c6eca59c75b5ec0a Mon Sep 17 00:00:00 2001
From: Petros Siligkounas <petross404@gmail.com>
Date: Wed, 13 Feb 2019 19:51:15 +0200
Subject: [PATCH] Handle Gentoo's LLVM directory layout

---
 src/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e38fbd1..d6865a3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -74,9 +74,16 @@ endif()
 string(REPLACE "svn" "" LLVM_VERSION "${LLVM_VERSION}")
 string(REGEX REPLACE "git.*$" "" LLVM_VERSION "${LLVM_VERSION}")
 if(NOT CLANG_BUILTIN_HEADERS_DIR)
-    set(CLANG_BUILTIN_HEADERS_DIR "${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION}/include")
+    #Gentoo Linux systems use a different directory layout
+    if(${CMAKE_BUILD_TYPE} MATCHES "Gentoo")
+	set(CLANG_BUILTIN_HEADERS_DIR "${LLVM_LIBRARY_DIR}/../include/clang-c")
+    else()
+	set(CLANG_BUILTIN_HEADERS_DIR "${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION}/include")
+    endif()
 endif()
+
 file(GLOB BUILTINS_HEADERS "${CLANG_BUILTIN_HEADERS_DIR}/*.h")
+
 if(NOT BUILTINS_HEADERS)
     message(FATAL_ERROR "Could not find any clang builtins headers in ${CLANG_BUILTIN_HEADERS_DIR}")
 endif()
