--- a/src/pcm/pcm_direct.h	2018-02-04 13:49:17.272885102 +0200
+++ b/src/pcm/pcm_direct.h	2018-02-04 13:50:05.968121886 +0200
@@ -229,7 +229,7 @@
 #define snd_pcm_direct_munmap \
 	snd1_pcm_direct_munmap
 #define snd_pcm_direct_prepare \
-	snd1_pcm_direct_prepare
+	__attribute__ ((used)) snd1_pcm_direct_prepare
 #define snd_pcm_direct_resume \
 	snd1_pcm_direct_resume
 #define snd_pcm_direct_timer_stop \
